### PR TITLE
Dynamic partition/url suffix references in cfn

### PR DESCRIPTION
**Pull request by austinbyers**
## Background

As we prepare to add support for AWS China for enterprise customers, the first step is to make sure all of the CloudFormation is partition-aware.

We were pretty good at this already, so it was only a handful of places that were missing these.

We can fully support AWS China for the community edition once the Beijing region adds support for Cognito user pools

## Changes

- Replace all `amazonaws.com` with `${AWS::URLSuffix}` references except for service principals
- Replace all `:aws:` references with `${AWS::Partition}`, unless it's actually part of a hardcoded name

## Testing

- `mage test:cfn`
- Enterprise china deployment
- Rigorous community deploy testing will happen as part of #2524

### CODE REVIEW

File: deployments/auxiliary/cloudformation/panther-log-analysis-iam.yml [hunk 1]
```diff
@@ -100,7 +100,7 @@ Resources:
               - Effect: Allow
                 Action: s3:GetBucketLocation
                 Resource: !Sub
-                  - 'arn:aws:s3:::${Bucket}'
+                  - arn:${AWS::Partition}:s3:::${Bucket}
```
Comment (s0l0ist) [inline deployments/auxiliary/cloudformation/panther-log-analysis-iam.yml:103]: Do we need the quotes put back?

File: internal/core/source_api/api/testdata/panther-log-analysis-iam-updated.yml [hunk 2]
```diff
@@ -100,7 +100,7 @@ Resources:
               - Effect: Allow
                 Action: s3:GetBucketLocation
                 Resource: !Sub
-                  - 'arn:aws:s3:::${Bucket}'
+                  - 'arn:${AWS::Partition}:s3:::${Bucket}'
```
Comment (s0l0ist) [inline internal/core/source_api/api/testdata/panther-log-analysis-iam-updated.yml:103]: If we don't need the quotes from my other comment, then maybe we should take them out here as well.

File: internal/core/source_api/api/testdata/panther-log-analysis-iam-updated.yml [hunk 3]
```diff
@@ -110,7 +110,7 @@ Resources:
               - Effect: Allow
                 Action: s3:GetObject
                 Resource: !Sub
-                  - 'arn:aws:s3:::${Bucket}/*'
+                  - 'arn:${AWS::Partition}:s3:::${Bucket}/*'
```
Comment (s0l0ist) [inline internal/core/source_api/api/testdata/panther-log-analysis-iam-updated.yml:113]: If we don't need the quotes ... remove them?

**Review by s0l0ist — APPROVED** [review]

File: deployments/auxiliary/cloudformation/panther-log-analysis-iam.yml (hunk 1 above)
Comment (austinbyers) [reply deployments/auxiliary/cloudformation/panther-log-analysis-iam.yml:103]: Nope, YAML strings generally don't need to be quoted. There are some exceptions (e.g. `'*'` or large numbers)

File: internal/core/source_api/api/testdata/panther-log-analysis-iam-updated.yml (hunk 2 above)
Comment (austinbyers) [reply internal/core/source_api/api/testdata/panther-log-analysis-iam-updated.yml:103]: Sure, good idea. This is just a test file, but might as well

**Pull request merged**